Netcdf to merge

geometer opened the pull request.

Commit:
a quick fix for wiggle displaying issue

Commit:
more debug info during wiggle processing

Commit:
sample <-> assay mapping from DB, not from netcdfs

Commit:
missimg imports

Commit:
one more wiggle quick fix ('_' are now encoded by '__')

Conflicts:

	atlas-web/src/main/webapp/scripts/experiment.js

Commit:
Merge branch 'develop' into netcdf

Conflicts:
	atlas-web/src/main/java/uk/ac/ebi/gxa/requesthandlers/wiggle/WiggleRequestHandler.java

Commit:
fixed access rights

Commit:
code simplification: uVALnum field in netcdf's is never used now

Commit:
uVALnum has been removed from NetCDFCreator

Commit:
cleanup: unused method has been removed

geometer committed:
Merge pull request #24 from gxa/updates

Incremental updates code dropped

Commit:
cleanup: access rights

Commit:
cleanup: no netcdf-specific exceptions outside of netcdf packages

Commit:
cleanup: access rights

Commit:
EFSC NetCDF variable is not used now

Commit:
Merge branch 'develop' into netcdf

Commit:
NetCDFCreator refactoring: experiment and arraydesign are now final fields

Commit:
NetCDFCreator code simplification

Commit:
imports re-arranging

Commit:
new class that will replace NetCDFProxy/NetCDFCreator calls

Commit:
repackaging

Commit:
module renaming: netcdf-reader => atlas-data-storage

Commit:
unused file has been removed

Commit:
repackaging

Commit:
the only way to create NetCDFCreator is to call AtlasNetCDFDAO.getNetCDFCreator method

Commit:
cleanup

Commit:
cleanup

Commit:
constructor NetCDFProxy(File) is not public now

Commit:
NetCDFDescriptor(File) constructor is not public now

Commit:
proxyId has gone

Commit:
Merge branch 'develop' into netcdf

Commit:
ArrayDesigh.equals problem (not a final solution)

Commit:
missing file

Commit:
enabled loading data from CEL files located in the same directory as IDF/SDRF files (http://bar.ebi.ac.uk:8080/trac/ticket/2993)

Commit:
enabled loading experiment data from the files in same directory where IDF/SDRF are located

Commit:
correct error message after failure in R code

Commit:
better loading problem notification

Commit:
Merge branch 'my-netcdf' into netcdf

Commit:
added createExperimentWithData method

Commit:
introduced AtlasDataException

Commit:
double pipeline split test is not actual now

Commit:
method getAllNetCDFs has gone

Commit:
getNetCDFDescriptors method has been moved into ExperimentWithData

Commit:
ExperimentalData is based on ExperimentWithData now

Commit:
netcdf refactoring: in progress

Commit:
ArrayDesign => ArrayDesignDecorator

Commit:
more code simplification

Commit:
code simplification

Commit:
NetCDFProxy usages will be replaced with ExperimentWithData usages

Commit:
NetCDFProxy -> ExperimentWithData

Commit:
NetCDFProxy -> ExperimentWithData

Commit:
no NetCDFProxy in API v1

Commit:
NetCDFProxy => ExperimentWithData (in progress)

Commit:
fixes for ExperimentalDataTest

Commit:
no NetCDFProxy in API v2

Commit:
Merge branch 'develop' into netcdf

Conflicts:
	atlas-loader/src/main/java/uk/ac/ebi/gxa/loader/steps/ArrayDataStep.java
	atlas-loader/src/main/resources/R/normalizeOneExperiment.R
	indexbuilder/src/main/java/uk/ac/ebi/gxa/index/builder/service/ExperimentAtlasIndexBuilderService.java
	indexbuilder/src/main/java/uk/ac/ebi/gxa/index/builder/service/GeneAtlasBitIndexBuilderService.java

Commit:
index builder refactoring: experiments instead of ncdfs

Commit:
NetCDF splitting code has been added

Commit:
Migrator interface

Commit:
Merge branch 'develop' into netcdf

Conflicts:
	atlas-loader/src/main/java/uk/ac/ebi/gxa/loader/service/AtlasNetCDFUpdaterService.java
	indexbuilder/src/main/java/uk/ac/ebi/gxa/index/builder/service/GeneAtlasBitIndexBuilderService.java

Commit:
Merge branch 'develop' into netcdf

Conflicts:
	atlas-web/src/main/webapp/WEB-INF/atlasApplicationContext.xml

Commit:
unused import

Commit:
fixed access rights

Commit:
unused (commented) code has been removed

Commit:
proxyId -> arrayDesign (local variable renaming)

Commit:
unused function has been removed

Commit:
Optimize imports, drop temporary variables

Commit:
old netcdf version name is supported

Commit:
Merge branch 'netcdf' of github.com:gxa/gxa into netcdf

Commit:
tests fixes

Commit:
netcdf file with version 0.0

Commit:
exceptions handling

Commit:
mistyping in comment

Commit:
consistent hashCode implementation

Commit:
compilation issue

Commit:
log.warn added

Commit:
Nonnull annotation

Commit:
comment added

Commit:
comment has been added

Commit:
migrator has been removed from this branch

alf239 commented:
On nightly (http://lime.ebi.ac.uk:14056/gxa/admin.html)

```

2011-07-29 14:06:59,462 INFO  [pool-7-thread-1] netcdf.NetCDFCreator (NetCDFCreator.java:851) - Writing NetCDF file to /****/E-MEXP-999_A-AFFY-25.nc2182490814731540537.tmp
2011-07-29 14:06:59,652 DEBUG [pool-7-thread-1] dao.AtlasDAO (AtlasDAO.java:153) - finishSession()
2011-07-29 14:06:59,653 ERROR [pool-7-thread-1] loader.DefaultAtlasLoader$1 (DefaultAtlasLoader.java:126) - Loading error
java.lang.ArrayIndexOutOfBoundsException
        at ucar.ma2.Index3D.setDim(Index3D.java:100)
        at ucar.ma2.Index.set(Index.java:664)
        at uk.ac.ebi.gxa.netcdf.NetCDFCreator.writeMap(NetCDFCreator.java:731)
        at uk.ac.ebi.gxa.netcdf.NetCDFCreator.writeSampleOntologies(NetCDFCreator.java:697)
        at uk.ac.ebi.gxa.netcdf.NetCDFCreator.write(NetCDFCreator.java:455)
        at uk.ac.ebi.gxa.netcdf.NetCDFCreator.createNetCdf(NetCDFCreator.java:855)
        at uk.ac.ebi.gxa.loader.service.AtlasNetCDFUpdaterService.writeNetCDF(AtlasNetCDFUpdaterService.java:149)
        at uk.ac.ebi.gxa.loader.service.AtlasNetCDFUpdaterService.process(AtlasNetCDFUpdaterService.java:65)
        at uk.ac.ebi.gxa.loader.DefaultAtlasLoader$1$1.process(DefaultAtlasLoader.java:104)
        at uk.ac.ebi.gxa.loader.UpdateNetCDFForExperimentCommand.visit(UpdateNetCDFForExperimentCommand.java:39)
        at uk.ac.ebi.gxa.loader.DefaultAtlasLoader$1.run(DefaultAtlasLoader.java:75)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:441)
        at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
        at java.util.concurrent.FutureTask.run(FutureTask.java:138)
        at java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:886)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:908)
        at java.lang.Thread.run(Thread.java:662)
```

```
2011-07-29 14:20:03,068 INFO  [pool-7-thread-1] netcdf.NetCDFCreator (NetCDFCreator.java:851) - Writing NetCDF file to /****/E-GEOD-911_A-AFFY-2.nc8690837628560164396.tmp
2011-07-29 14:20:03,174 DEBUG [pool-7-thread-1] dao.AtlasDAO (AtlasDAO.java:153) - finishSession()
2011-07-29 14:20:03,174 ERROR [pool-7-thread-1] loader.DefaultAtlasLoader$1 (DefaultAtlasLoader.java:126) - Loading error
java.lang.ArrayIndexOutOfBoundsException
        at ucar.ma2.Index3D.setDim(Index3D.java:100)
        at ucar.ma2.Index.set(Index.java:664)
        at uk.ac.ebi.gxa.netcdf.NetCDFCreator.writeMap(NetCDFCreator.java:731)
        at uk.ac.ebi.gxa.netcdf.NetCDFCreator.writeAssayOntologies(NetCDFCreator.java:702)
        at uk.ac.ebi.gxa.netcdf.NetCDFCreator.write(NetCDFCreator.java:454)
        at uk.ac.ebi.gxa.netcdf.NetCDFCreator.createNetCdf(NetCDFCreator.java:855)
```

alf239 commented:
Also, there's an old term-vs-accession problem. Did we test this functionality?

Commit:
`getTerm()` always returns `null` — use `getAccession()` instead

Commit:
use explicit `getAccession()` instead of implicit `toString()`

Commit:
Dropping the `term` field as unused and obsolete

alf239 commented:
Fixed the NPE; please check out the AIOOBE problems. By the way, do we really need to proceed with ontology mappings in NetCDFs?

alf239 commented:
Regarding the E-MEXP-999, the problem seems to be caused by non-unique properties:

``` sql
select s.sampleid, p.name as property, pv.name as value from a2_samplepv spv
join a2_propertyvalue pv on spv.propertyvalueid= pv.propertyvalueid
join a2_property p on pv.propertyid= p.propertyid
join a2_sample s on spv.sampleid = s.sampleid 
join a2_experiment e on e.experimentid = s.experimentid
where e.accession = 'E-MEXP-999'
order by s.sampleid;
```

<table>
<tr><th>SAMPLEID</th><th>PROPERTY</th><th>VALUE</th></tr>
<tr><td>240222912   </td><td>age</td><td>   20 days</td></tr>
<tr><td>240222912   </td><td>developmental_stage</td><td>   immature</td></tr>
<tr><td>240222912   </td><td>initial_time_point</td><td>    birth</td></tr>
<tr><td>240222912   </td><td>organism_part</td><td> uterus</td></tr>
<tr><td>240222912</td><td>  organism_part</td><td>  ovary</td></tr>
<tr><td>240222912</td><td>  sex</td><td>    female</td></tr>
</table>

Commit:
`getPropertyNames()` returns `SortedSet` now

So that we could avoid the situation when length ot propertyNames list is not equal to length of property Multimap keySet
